Core/Core/Extensions/UIImageViewExtensions.swift

```diff
@@ -156,36 +156,39 @@ public class ImageLoader {
         var loaded: LoadedImage?
         if let image = image {
             loaded = LoadedImage(image: image, repeatCount: repeatCount)
+            ImageLoader.rendered[key] = loaded
+        }
+        for loader in ImageLoader.loading[key] ?? [] {
+            loader.view?.load(url: url, didCompleteWith: loaded, error: error)
         }
-        view?.load(url: url, didCompleteWith: loaded, error: error)
+        ImageLoader.loading[key] = nil
     }
 
     // MARK: - SVG snapshot
 
-    @available(iOS 11.0, *)
     func svgFrom(data: Data) {
-        let view = WKWebView(frame: frame, configuration: WKWebViewConfiguration())
-        view.isOpaque = false
+        let config = WKWebViewConfiguration()
+        config.websiteDataStore = .nonPersistent()
+        let view = WKWebView(frame: frame, configuration: config)
         view.backgroundColor = .clear
-        view.scrollView.backgroundColor = .clear
+        view.isOpaque = false
         webView = view
 
-        observation = view.observe(\.title, options: .new) { webView, _ in
-            guard webView.title == "SVG" else { return }
-            webView.takeSnapshot(with: nil) { snapshot, error in
-                self.webView = nil
-                self.observation = nil
-                self.handle(snapshot, 0, error)
-            }
-        }
-
         let dataUri = "data:image/svg+xml;base64,\(data.base64EncodedString())"
         let html = """
             <!doctype html>
             <meta name="viewport" content="width=device-width,initial-scale=1,user-scalable=no">
-            <style>html{height:100%;background:no-repeat \(cssFromContentMode(contentMode)) url(\(dataUri))}</style>
-            <script>window.onload=()=>{setTimeout(()=>{document.title="SVG"},0)}</script>
+            <style>*{margin:0;padding:0;height:100%;width:100%;}</style>
+            <img src="\(dataUri)" style="object-position:center;object-fit:contain;"
+                onload="setTimeout(()=>webkit.messageHandlers.svg.postMessage(''),17)"
```

Comment:
`onload` on an image is more specific than on window, which may not include background images. The timeout is sadly necessary because WK renders the sizing wrong in the first pass.